app/connector/sql/src/main/java/io/syndesis/connector/sql/common/DbAdapter.java

```diff
@@ -0,0 +1,56 @@
+package io.syndesis.connector.sql.common;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import io.syndesis.connector.sql.db.Db;
+import io.syndesis.connector.sql.db.DbDerby;
+import io.syndesis.connector.sql.db.DbMysql;
+import io.syndesis.connector.sql.db.DbOracle;
+import io.syndesis.connector.sql.db.DbPostgresql;
+import io.syndesis.connector.sql.db.DbStandard;
+
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+public class DbAdapter {
```

Comment:
Perhaps instead of `DbAdapter`, `Db` instance can be obtained straight from the `DbEnum`, I'm thinking we add a `DbEnum` and then we have to remember to update the code here also.

So `DbEnum` can have a method like `Db fetchAdapterFor(Connection con)` and it could hold a `db` field with the singleton instance of `Db`.

Would call the `Db` just `DatabaseAdapter` or similar...

Reply:
Y I actually want to get rid of the DbEnum altogether in the follow up refactoring PR.

Reply:
Awesome, even better...